lib/postgres_to_redshift/keys.rb

```diff
@@ -0,0 +1,49 @@
+module PostgresToRedshift
+  class Keys
+    def initialize(source_connection:)
+      @source_connection = source_connection
+    end
+
+    def all
+      all_keys
+    end
+
+    private
+
+    attr_reader :source_connection
+
+    def redshift_include_tables
+      @redshift_include_tables ||= ENV['REDSHIFT_INCLUDE_TABLES'].split(',')
+    end
+
+    def all_keys
+      source_connection.exec(find_all_keys_sql).map do |key_attributes|
+        Key.new(attributes: key_attributes)
+      end.compact
+    end
+
+    def find_all_keys_sql
+      sql = <<~SQL
+        select
+          conrelid::regclass AS table_name,
+          conname as name,
+          pg_get_constraintdef(oid) as key_definition,
+          contype as key_type
+        from pg_constraint
+        where connamespace = 'public'::regnamespace
+      SQL
+
+      if ENV['REDSHIFT_INCLUDE_TABLES']
```

Comment:
Could this be simplified based on this comment? https://github.com/meetcleo/postgres_to_redshift/pull/8/files#r325732075